### ISSUE TITLE
func (bsm *blockStreamMerger) Merge use goto, it is really ugly.

### ISSUE THREAD

**Issue by ahfuzhang** [VictoriaMetrics/VictoriaMetrics#2248]
see: https://github.com/VictoriaMetrics/VictoriaMetrics/blob/99ee1c13c05479d974d496c8b44c5c6b84ed03b0/lib/mergeset/merge.go#L103

I think we can use for {} to instead of goto.

valyala, you are my YYDS.
Pls don't let beginner ruin your niubility talent. :-)

thanks.

**Comment by valyala** [comment]
Agreed that this `goto` usage looks ugly. But I couldn't figure out less ugly code without `goto` there :( @ahfuzhang , feel free sending a pull request, which improves the readability and maintainability of this code.
